app/models/classification.rb

```diff
@@ -522,17 +540,22 @@ def self.tag2human(tag)
 
   private_class_method :tag2human
 
+  def derive_tag_name
+    Classification.name2tag(name, category? ? 0 : parent, ns)
```

Comment:
Specifically, wouldn't `Classification.name2tag(name, parent, ns)` work?

Reply:
with `parent_id=0`, sending in `parent` would do a lookup (and it would convert to nil) and we would be confused on the other side.

I suppose I could pass in a `nil` and change it to properly work a `nil` rather than the current implementation.

Reply:
Ah right, the usual idiom is `name2tag(name, parent_id, ns)`.  Then name2tag does the lookup conditionally on parent_id.  But I suppose that sometimes costs an unnecessary lookup when `parent` object is already loaded.

On the positive side, this is the high-level "friendlier name2tag" method I'd previously wished we had :+1:
No objection to keeping a tiny complication to do it optimally.

Reply:
@cben thanks so much for pressing on this - I have revisited and hope I found a better solution